### PR TITLE
Extract config

### PR DESCRIPTION
**Pull request by darkspirit510**
First step on the "reduce BenchmarkScore size" path 😅 I extracted reading/parsing of configuration file to own class. For now, I keep the static access (although I want to change this in the future, too). I slightly reduced the logging (why print the name that you just requested to be read).

### CODE REVIEW

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java [hunk 1]
```diff
@@ -98,42 +79,16 @@ public class BenchmarkScore extends AbstractMojo {
...
-    private static String focus = "none";
-    // This is used to indicate that results from multiple versions of a test suite are included in
-    // these results. Each set in their own directory with their associated expectedresults file.
-    public static boolean mixedMode = false;
-
-    private static String expectedResultsFileName;
-    private static String resultsFileOrDirName;
-
     // Customizations of these can be set via scorecardconfig.yaml files
     // This affects details within whatever type of scorecards are being generated
```
Comment (davewichers) [inline plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:84]: Doesn't this comment either need to be updated or moved as it seems to be left hanging there without context.

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java [hunk 2]
```diff
@@ -163,140 +118,47 @@ public class BenchmarkScore extends AbstractMojo {
...
-        InputStream yamlFileStream = null;
-        if (args == null || args.length == 0) {
-            // No arguments is OK
-        } else if (args.length != 0 && args.length != 2) {
-            System.out.println(USAGE_MSG);
-        } else if (args.length == 2) {
+    private static void loadConfigFromCommandLineArguments(String[] args) {
+        if (args == null || args.length != 2) {
+            config = Configuration.fromDefaultConfig();
+        } else {
```
Comment (davewichers) [inline plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:131]: Its not clear to me that this handles all the error cases properly. If args is != null and args.length is != 2 we should display the usage error. But it looks like in that case it just loads the default policy without showing an error.

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java [hunk 3]
```diff
@@ -163,140 +118,47 @@ public class BenchmarkScore extends AbstractMojo {
...
-                // This should never happen
-                System.out.println(
-                        "Couldn't determine URI of existing resource: '"
-                                + DEFAULTCONFIGFILE
-                                + "' for some reason.");
-                e.printStackTrace();
-            }
+        if (!config.includePrecision) {
+            // This two values are both included or not included together (currently)
+            PRECISIONKEYENTRY = "";
```
Comment (davewichers) [inline plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:155]: Should be 'These two values ...

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 1 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:84]: Oh yeah, missed them. Will move them this weekend.

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 2 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:131]: I'll add some test cases to make sure that there is an error message in the expected cases!

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 3 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:155]: Weird, this is not my comment, but I can fix it

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 1 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java]: Done

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 3 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:164]: Done

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 2 above)
Comment (davewichers) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:140]: So you've added some test cases/made changes to make sure there are proper error messages in all cases?

**Comment by davewichers** [comment]
I think this one is ready to be merged. Do you agree?

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 2 above)
Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:141]: I added the message, but did not write any test. Not sure, if this is an issue, currently time is a limiting factor to me 🙈

**Comment by darkspirit510** [comment]
See comment above. If this isn't an issue, this PR can be merged.

File: plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java (hunk 2 above)
Comment (davewichers) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:141]: @darkspirit510 - any chance you can write a few test cases for this before I merge?

Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:141]: If you prefer, of course!

Comment (darkspirit510) [reply plugin/src/main/java/org/owasp/benchmarkutils/score/BenchmarkScore.java:143]: please check the tests 😊

**Pull request merged**